comp/metadata/runner/runnerimpl/runner.go

```diff
@@ -137,12 +176,21 @@ func (r *runnerImpl) handleProvider(p MetadataProvider) {
 // start is called by FX when the application starts. Lifecycle hooks are blocking and called sequencially. We should
 // not block here.
 func (r *runnerImpl) start() error {
-	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers))
-	for _, optionaP := range r.providers {
-		if p, isSet := optionaP.Get(); isSet {
-			go r.handleProvider(p)
+	r.log.Debugf("Starting metadata runner with %d providers", len(r.providers)+len(r.priorityProviders))
+
+	go func() {
```

Comment:
By wrapping the functionality of initializing the provides we ensure not blocking the application start.

Reply:
We should add a comment explaining this.

Reply:
Should the comment on the `start` not be enough? 
```
// start is called by FX when the application starts. Lifecycle hooks are blocking and called sequencially. We should
// not block here.
```